### DISCUSSION TITLE
Define "background processing" in README, documentation and website

### DISCUSSION THREAD

**Discussion by tigregalis** [ayrat555/fang#102] [General]
Define "background processing" in README, documentation and website.

Consider linking to, and/or incorporating content from, https://fang.badykov.com/blog/async-processing/

As someone new to systems and desktop/server application programming, I didn't properly understand the term "background processing" in this context (I only had a notion of what it might be). The readme and documentation currently assumes the reader already understands what "background processing" is and what it is for.

And it turns out you've already got a good definition in your first blog post.
> In software engineering background processing is a common approach for solving several problems:
> - Carry out periodic tasks. For example, deliver notifications, update cached values.
> - Defer expensive work so your application stays responsive while performing calculations in the background

It might be that perhaps if I don't understand the term I'm not the audience, however, now that I do understand what it's for, I can see myself using it. I think spelling it out would also be beneficial to developers whose first language isn't English as well.

**Comment by pxp9** [comment]
Thank you for your suggestion @tigregalis, we will consider to create in our blog guidelines to learn:

- what is background processing ?

- How to create a queue? 

- How to insert tasks in a queue ? 

- How to start workers to execute tasks ?
